### PR TITLE
Allow to configure custom serializer groups

### LINKED ISSUE #238: Allow configurable serialization group(s)
Currently the serialization groups support requires a group of 'searchable' (Searchable::NORMALIZATION_GROUP).  

Since I already have a serialization group set up on my entities, I'd like to use that.  Actually, I'd like to use more than one, so that I can say something like

```yaml
    indices:
        - name: movies
          class: App\Entity\Movie
          enable_serializer_groups: true
          serializer_groups: ['movie.read','movie.export']
```

On an unrelated note, meilisearch rocks!  I can't wait to start using it in production.

### PR DESCRIPTION
**Pull request by norkunas**
# Pull Request

## Related issue
Fixes #238

## What does this PR do?
- Allows to configure custom serializer groups instead of the default one `Meilisearch\Bundle\Searchable::NORMALIZATION_GROUP`.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

### CODE REVIEW

**Comment by bidoubiwa** [comment]
Hey @norkunas! Thanks a lot for your contribution 🔥

The maintainer of this repo @brunoocasali is currently on holiday. He'll be back next week to review you!

File: src/SearchableEntity.php
Comment (norkunas) [inline src/SearchableEntity.php:53]: this could be some sort of a BC break, but will leave it to you to decide if we need a compatibility here

**Comment by brunoocasali** [comment]
Hi, @norkunas. I will need more time to review this PR. Sorry about that!

File: tests/config/meilisearch.yaml
Comment (brunoocasali) [inline tests/config/meilisearch.yaml:42]: I tried to change to `serializer_groups: ['public']` but the tests did not break, I was expecting to have a breaking in the last part:

File: tests/Integration/CommandsTest.php
Comment (brunoocasali) [inline tests/Integration/CommandsTest.php:415]: See the comment in the meilisearch.yaml

File: src/DependencyInjection/Configuration.php
Comment (brunoocasali) [inline src/DependencyInjection/Configuration.php:50]: Should it be `serializer_groups`? 
```suggestion
                                ->info('When set to true, it will call normalize method with an extra groups parameter "serializer_groups" => [Searchable::NORMALIZATION_GROUP]')
```

If yes, I still see `groups` on `src/SearchableEntity.php` (but I don't know if they are related or not)

**Review by brunoocasali — CHANGES_REQUESTED** [review]
Hi @norkunas very sorry for taking too much time to answer you.

So, I tried to make some changes locally and the tests did not break, can you help me understand more about the change you are proposing?

Thank you!

**Comment by norkunas** [comment]
> Hi @norkunas very sorry for taking too much time to answer you.
> 
> So, I tried to make some changes locally and the tests did not break, can you help me understand more about the change you are proposing?
> 
> Thank you!

Will check it on monday. See the issue to understand about it :)

File: src/SearchableEntity.php
Comment (brunoocasali) [reply src/SearchableEntity.php:53]: I don't think we need to keep compatibility, but is the change just from `useSerializerGroup` to `normalizationGroups`? and both are arrays right?

Comment (norkunas) [reply src/SearchableEntity.php:53]: `useSerializerGroup` was a boolean

File: tests/config/meilisearch.yaml
Comment (norkunas) [reply tests/config/meilisearch.yaml:42]: Did you cleared the cache? Because after removing `private` group I got:
```
1) Meilisearch\Bundle\Tests\Integration\CommandsTest::testImportsDummyWithCustomGroups
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
         'objectID' => 1
         'id' => 1
         'name' => 'Dummy 1'
-        'createdAt' => '2024-04-04T07:32:01+00:00'
     )
     1 => Array &2 (
         'objectID' => 2
         'id' => 2
         'name' => 'Dummy 2'
-        'createdAt' => '2024-04-04T07:32:02+00:00'
     )
 )
```

Comment (norkunas) [reply tests/config/meilisearch.yaml:42]: I think bundle should take care of clearing the cache before running locally, because I always need to clear it manually so changes are taken into account

File: composer.json
Comment (norkunas) [inline composer.json:69]: Symfony phpunit-bridge is in dev dependencies, but was not leveraged

File: phpunit.xml.dist
Comment (norkunas) [inline phpunit.xml.dist:4]: better to reference the local schema that is downloaded with phpunit when using symfony phpunit-bridge

File: phpunit.xml.dist
Comment (norkunas) [inline phpunit.xml.dist:2]: not it will always take care to remove the cache before running tests

File: src/DependencyInjection/Configuration.php
Comment (norkunas) [reply src/DependencyInjection/Configuration.php:50]: > If yes, I still see groups on src/SearchableEntity.php (but I don't know if they are related or not)
That's because Symfony uses just `groups` both for normalization and denormalization

File: phpunit.xml.dist
Comment (norkunas) [inline phpunit.xml.dist:17]: this does not run in travis

File: phpunit.xml.dist
Comment (norkunas) [inline phpunit.xml.dist:18]: deprecation logging must not be disabled!

File: tests/config/config.yaml
Comment (norkunas) [inline tests/config/config.yaml:4]: Symfony deprecated not setting this

File: tests/config/meilisearch.yaml
Comment (brunoocasali) [reply tests/config/meilisearch.yaml:42]: I think that was the issue, I tried your new version with the auto caching clear and it is working!

**Comment by brunoocasali** [comment]
> Okay so now there are many unrelated changes, if you'll want I could move them to another PR

not needed @norkunas!

**Review by brunoocasali — APPROVED** [review]
Thanks a lot for the contribution @norkunas 🎉 

bors merge

**Review by brunoocasali — APPROVED** [review]
Thanks a lot for the contribution @norkunas 🎉 

bors merge

**Pull request merged**
